src/modules/job-manager/plugins/perilog.c

```diff
@@ -191,27 +192,41 @@ static void emit_finish_event (struct perilog_proc *proc,
          *   the exception is raised:
          */
         if (status != 0 && !proc->canceled) {
+            flux_t *h = flux_jobtap_get_flux (proc->p);
             int code = WIFEXITED (status) ? WEXITSTATUS (status) : -1;
             int sig;
             char *errmsg;
+            char *hosts = NULL;
 
+            if (!(hosts = flux_hostmap_lookup (h, proc->failed_ranks, NULL)))
+                hosts = strdup ("unknown");
+
+            if (proc->timedout) {
+                rc = asprintf (&errmsg,
+                               "prolog timed out on %s (rank %s)",
+                               hosts,
+                               proc->failed_ranks);
+            }
             /*  Report that prolog was signaled if WIFSIGNALED() is true, or
              *  exit code > 128 (where standard exit code is 127+signo from
              *  most shells)
              */
-            if (WIFSIGNALED (status) || code > 128) {
+            else if (WIFSIGNALED (status) || code > 128) {
                 sig = WIFSIGNALED (status) ? WTERMSIG (status) : code - 128;
                 rc = asprintf (&errmsg,
-                               "prolog killed by signal %d%s",
+                               "prolog killed by signal %d on %s (rank %s)",
                                sig,
-                               sig == SIGTERM ?
-                               " (timeout or job canceled)" :
-                               "");
+                               hosts,
```

Comment:
`hosts` could be null if both `flux_hostmap_lookup()` and `strdup()` fail

Reply:
and printf will report "(null)" -- would `"unknown"` be better.

Reply:
sorry, I keep forgetting that we decided that is portably safe now.  FIne as is then.